node/sc/main_bridge_handler.go

```diff
@@ -103,26 +102,32 @@ func (mbh *MainBridgeHandler) handleCallMsg(p BridgePeer, msg p2p.Msg) error {
 // handleServiceChainTxDataMsg handles service chain transactions from child chain.
 // It will return an error if given tx is not TxTypeChainDataAnchoring type.
 func (mbh *MainBridgeHandler) handleServiceChainTxDataMsg(p BridgePeer, msg p2p.Msg) error {
-	//pm.txMsgLock.Lock()
+	// pm.txMsgLock.Lock()
 	// Transactions can be processed, parse all of them and deliver to the pool
 	var txs []*types.Transaction
 	if err := msg.Decode(&txs); err != nil {
 		return errResp(ErrDecode, "msg %v: %v", msg, err)
 	}
 
 	// Only valid txs should be pushed into the pool.
-	validTxs := make([]*types.Transaction, 0, len(txs))
-	//validTxs := []*types.Transaction{}
-	var err error
-	for i, tx := range txs {
+	// Invalid txs found are sent to child chain for further action.
+	var invalidTxs []InvalidParentChainTx
+	for _, tx := range txs {
 		if tx == nil {
-			err = errResp(ErrDecode, "tx %d is nil", i)
+			invalidTxs = append(invalidTxs, InvalidParentChainTx{tx.Hash(), errResp(ErrDecode, "tx is nil").Error()})
 			continue
 		}
-		validTxs = append(validTxs, tx)
+		if err := mbh.mainbridge.txPool.AddRemote(tx); err != nil {
```

Comment:
Is there any reason to double check invalidtx in the 116th if statement and in the 120th if statement? I thought that if I check only the 116th if statement, all errors would come out.

Reply:
The first condition is for decoding error and the second one is for rejected tx error such as lower gasPrice.